cmd/skoap/main.go

```diff
@@ -0,0 +1,318 @@
+	fs.BoolVar(&insecure, insecureFlag, false, insecureUsage)
+	fs.StringVar(&authUrlBase, authUrlBaseFlag, "", authUrlBaseUsage)
+	fs.StringVar(&teamUrlBase, teamUrlBaseFlag, "", teamUrlBaseUsage)
+	fs.StringVar(&certPathTLS, tlsCertFlag, "", certPathTLSUsage)
+	fs.StringVar(&keyPathTLS, tlsKeyFlag, "", keyPathTLSUsage)
+	fs.BoolVar(&verbose, verboseFlag, false, verboseUsage)
+	fs.BoolVar(&experimentalUpgrade, experimentalUpgradeFlag, false, experimentalUpgradeUsage)
+
+	err := fs.Parse(os.Args[1:])
+	if err != nil {
+		if err == flag.ErrHelp {
+			os.Exit(0)
+		}
+
+		os.Exit(-1)
+	}
+}
+
+func logUsage(message string) {
+	fmt.Fprintf(os.Stderr, "%s\n", message)
+	os.Exit(-1)
+}
+
+func main() {
+	if verbose {
+		logrus.SetLevel(logrus.DebugLevel)
+	} else {
+		logrus.SetLevel(logrus.WarnLevel)
+	}
+
+	if targetAddress == "" && routesFile == "" && etcdUrls == "" {
+		logUsage("either the target address, a routes file or etcd urls needs to be specified")
+	}
+
+	// check that only one of targetAddress, routesFile and etcdUrls is
+	// defined
+	defined := 0
+	for _, f := range []string{targetAddress, routesFile, etcdUrls} {
+		if f != "" {
+			defined++
+		}
+	}
+
+	if defined > 1 {
+		logUsage("can only set one of: target address, etcd urls, or a routes file")
+	}
+
+	singleRouteMode := targetAddress != ""
+
+	if !singleRouteMode && (preserveHeader || realm != "" || scopes != "" || teams != "" || audit || auditBody != 1024) {
+		logUsage("the preserve-header, realm, scopes, teams, audit-log and audit-log-limit flags can be used only together with the target-address flag (single route mode)")
+	}
+
+	if !audit && auditBody != 1024 {
+		logUsage("the audit-log-limit flag can be set only together with the audit-log flag")
+	}
+
+	if scopes != "" && teams != "" {
+		logUsage("the scopes and teams flags cannot be used together")
+	}
+
+	teamCheckMode := teams != ""
+
+	if authUrlBase == "" {
+		authUrlBase = defaultAuthUrlBase
+	}
+
+	if teamUrlBase == "" {
+		teamUrlBase = defaultTeamUrlBase
+	}
+
+	o := skipper.Options{
+		Address:    address,
+		EtcdPrefix: etcdPrefix,
+		CustomFilters: []filters.Spec{
+			skoap.NewAuth(authUrlBase),
+			skoap.NewAuthTeam(authUrlBase, teamUrlBase),
+			skoap.NewBasicAuth(),
+			skoap.NewAuditLog(os.Stderr)},
+		AccessLogDisabled:   true,
+		ProxyOptions:        proxy.OptionsPreserveOriginal,
+		CertPathTLS:         certPathTLS,
+		KeyPathTLS:          keyPathTLS,
+		ExperimentalUpgrade: experimentalUpgrade,
+	}
+
+	if insecure {
+		o.ProxyOptions |= proxy.OptionsInsecure
+	}
+
+	if routesFile != "" {
+		o.RoutesFile = routesFile
+	} else if etcdUrls != "" {
+		var eus []string
+		if len(etcdUrls) > 0 {
+			eus = strings.Split(etcdUrls, ",")
+		}
+		o.EtcdUrls = eus
+	} else {
+		var filterArgs []interface{}
+		if realm != "" {
+			filterArgs = append(filterArgs, realm)
+		}
+
+		args := scopes
+		name := skoap.AuthName
+		if teamCheckMode {
+			args = teams
```

Comment:
not even fully sure if we should keep this separate command. Does it provide that much value?